lib/puppet/type/firewalld_custom_service.rb

```diff
@@ -0,0 +1,154 @@
+require 'puppet'
+
+Puppet::Type.newtype(:firewalld_custom_service) do
+  desc <<-DOC
+  @summary Creates a custom firewalld service.
+
+
+  You will still need to create a `firewalld_service` resource to bind your new
+  service to a zone.
+
+  @example Creating a custom 'test' service
+    firewalld_custom_service {'test':
+        ensure  => present,
+        ports   => [{'port' => '1234', 'protocol' => 'tcp'}]
+    }
+  DOC
+
+  ensurable do
+    defaultvalues
+    defaultto(:present)
+  end
+
+  newparam(:name, namevar: :true) do
+    desc 'The "short" name of the resource'
```

Comment:
I'm still unconvinced that `short` shouldn't be its own property. The name of the service, as returned by `--get-services` matches the filenames, not the `short` attribute.

```
[root@test ~]# firewall-cmd --get-services
RH-Satellite-6 amanda-client amanda-k5-client amqp amqps apcupsd audit bacula bacula-client bgp bitcoin bitcoin-rpc bitcoin-testnet bitcoin-testnet-rpc ceph ceph-mon cfengine check-mk-agent condor-collector ctdb dhcp dhcpv6 dhcpv6-client distcc dns docker-registry docker-swarm dropbox-lansync elasticsearch etcd-client etcd-server finger freeipa-ldap freeipa-ldaps freeipa-replication freeipa-trust ftp ganglia-client ganglia-master git gre high-availability http https imap imaps ipp ipp-client ipsec irc ircs iscsi-target isns jenkins kadmin kerberos kibana klogin kpasswd kprop kshell ldap ldaps libvirt libvirt-tls lightning-network llmnr managesieve matrix mdns minidlna mongodb mosh mountd mqtt mqtt-tls ms-wbt mssql murmur mysql nfs nfs3 nmea-0183 nrpe ntp nut openvpn ovirt-imageio ovirt-storageconsole ovirt-vmconsole plex pmcd pmproxy pmwebapi pmwebapis pop3 pop3s postgresql privoxy proxy-dhcp ptp pulseaudio puppetmaster quassel radius redis rpc-bind rsh rsyncd rtsp salt-master samba samba-client samba-dc sane sip sips slp smtp smtp-submission smtps snmp snmptrap spideroak-lansync squid ssh steam-streaming svdrp svn syncthing syncthing-gui synergy syslog syslog-tls telnet tftp tftp-client tinc tor-socks transmission-client upnp-client vdsm vnc-server wbem-http wbem-https wsman wsmans xdmcp xmpp-bosh xmpp-client xmpp-local xmpp-server zabbix-agent zabbix-server

[root@test ~]# ls /usr/lib/firewalld/services
amanda-client.xml        dhcp.xml                 https.xml         ldap.xml               nmea-0183.xml             puppetmaster.xml     snmptrap.xml             vdsm.xml
amanda-k5-client.xml     distcc.xml               http.xml          libvirt-tls.xml        nrpe.xml                  quassel.xml          snmp.xml                 vnc-server.xml
amqps.xml                dns.xml                  imaps.xml         libvirt.xml            ntp.xml                   radius.xml           spideroak-lansync.xml    wbem-https.xml
amqp.xml                 docker-registry.xml      imap.xml          lightning-network.xml  nut.xml                   redis.xml            squid.xml                wbem-http.xml
apcupsd.xml              docker-swarm.xml         ipp-client.xml    llmnr.xml              openvpn.xml               RH-Satellite-6.xml   ssh.xml                  wsmans.xml
audit.xml                dropbox-lansync.xml      ipp.xml           managesieve.xml        ovirt-imageio.xml         rpc-bind.xml         steam-streaming.xml      wsman.xml
bacula-client.xml        elasticsearch.xml        ipsec.xml         matrix.xml             ovirt-storageconsole.xml  rsh.xml              svdrp.xml                xdmcp.xml
bacula.xml               etcd-client.xml          ircs.xml          mdns.xml               ovirt-vmconsole.xml       rsyncd.xml           svn.xml                  xmpp-bosh.xml
bgp.xml                  etcd-server.xml          irc.xml           minidlna.xml           plex.xml                  rtsp.xml             syncthing-gui.xml        xmpp-client.xml
bitcoin-rpc.xml          finger.xml               iscsi-target.xml  mongodb.xml            pmcd.xml                  salt-master.xml      syncthing.xml            xmpp-local.xml
bitcoin-testnet-rpc.xml  freeipa-ldaps.xml        isns.xml          mosh.xml               pmproxy.xml               samba-client.xml     synergy.xml              xmpp-server.xml
bitcoin-testnet.xml      freeipa-ldap.xml         jenkins.xml       mountd.xml             pmwebapis.xml             samba-dc.xml         syslog-tls.xml           zabbix-agent.xml
bitcoin.xml              freeipa-replication.xml  kadmin.xml        mqtt-tls.xml           pmwebapi.xml              samba.xml            syslog.xml               zabbix-server.xml
ceph-mon.xml             freeipa-trust.xml        kerberos.xml      mqtt.xml               pop3s.xml                 sane.xml             telnet.xml
ceph.xml                 ftp.xml                  kibana.xml        mssql.xml              pop3.xml                  sips.xml             tftp-client.xml
cfengine.xml             ganglia-client.xml       klogin.xml        ms-wbt.xml             postgresql.xml            sip.xml              tftp.xml
condor-collector.xml     ganglia-master.xml       kpasswd.xml       murmur.xml             privoxy.xml               slp.xml              tinc.xml
ctdb.xml                 git.xml                  kprop.xml         mysql.xml              proxy-dhcp.xml            smtp-submission.xml  tor-socks.xml
dhcpv6-client.xml        gre.xml                  kshell.xml        nfs3.xml               ptp.xml                   smtps.xml            transmission-client.xml
dhcpv6.xml               high-availability.xml    ldaps.xml         nfs.xml                pulseaudio.xml            smtp.xml             upnp-client.xml
```

Picking a couple of those services and getting their short descriptions.
```
[root@test ~]# firewall-cmd --permanent --service bacula-client --get-short
Bacula Client
[root@splunk-test ~]# firewall-cmd --permanent --service high-availability --get-short
Red Hat High Availability
```

Reply:
@alexjfisher Ack...you're right. I was reading the output backwards at one point. Will update.